seaborn/categorical.py

```diff
@@ -1129,15 +1168,228 @@ def plot(self, ax, kws):
             ax.invert_yaxis()
 
 
-class _SwarmPlotter(_BoxPlotter):
+class _SwarmPlotter(_CategoricalScatterPlotter):
 
-    def __init__(self):
+    def __init__(self, x, y, hue, data, order, hue_order,
+                 split, orient, color, palette):
+        """Initialize the plotter."""
+        self.establish_variables(x, y, hue, data, orient, order, hue_order)
+        self.establish_colors(color, palette, 1)
 
-        pass
+        # Set object attributes
+        self.split = split
+        self.width = .8
 
-    def plot(self, ax):
+    def overlap(self, xy_i, xy_j, d):
+        """Return True if two circles with the same diameter will overlap."""
+        x_i, y_i = xy_i
+        x_j, y_j = xy_j
+        return np.linalg.norm([x_i - x_j, y_i - y_j]) < d
 
-        pass
+    def could_overlap(self, xy_i, swarm, d):
+        """Return a list of all swarm points that could overlap with target.
+
+        Assumes that we are working through a sorted storm.
+        """
+        _, y_i = xy_i
+        neighbors = []
+        for xy_j in swarm:
+            _, y_j = xy_j
+            if (y_i - y_j) < d:
+                neighbors.append(xy_j)
```

Comment:
No, because `swarm` in this function is only the points that have already been plotted, so working through a sorted list means that you'll never actually encounter a `y_j` that is greater than the `y_i`.